synapse/handlers/auth.py

```diff
@@ -892,6 +894,17 @@ def _do_validate_hash():
         else:
             return defer.succeed(False)
 
+    def _check_mau_limits(self):
+        """
+        Ensure that if mau blocking is enabled that invalid users cannot
+        log in.
+        """
+        if self.hs.config.limit_usage_by_mau is True:
+            current_mau = self.store.count_monthly_users()
```

Comment:
This method will be hit very often. So you might want to either add some cache on `store.count_monthly_users()` or use a value which might be generated by [generate_monthly_active_users](https://github.com/matrix-org/synapse/pull/3630/files#diff-01aac508ef38061957e6fb961582e5ceR521)

Reply:
agreed though deliberately omitted since count_monthly_users() is really only a stepping stone method to get the rest of the infra out. In current form only called regularly in the case where the config is enabled.